### PR TITLE
Don't create description links for self-referencing work packages.

### PR DESCRIPTION
**Pull request by crijke**
https://www.openproject.org/issues/165

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -734,7 +734,8 @@ def parse_redmine_links(text, project, obj, attr, only_path, options)
           end
         elsif sep == '###'
           oid = identifier.to_i
-          if work_package = WorkPackage.visible.find_by_id(oid, :include => :status)
+          work_package = WorkPackage.visible.find_by_id(oid, :include => :status)
+          if obj && work_package && obj.id != work_package.id
```
Comment (ulferts) [inline app/helpers/application_helper.rb]: What Do you do, when I do this:
- two tickets (A and B)
- A's description set to "###[B's id]"
- B's description set to "###[A's id]"

File: spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -317,6 +317,34 @@
...
+      context "Cyclic Description Links" do
+        let(:issue2) { FactoryGirl.create :work_package,
+                                         :project => project,
+                                         :author => project_member,
+                                         :type => project.types.first }
+
+        before do
+          issue2.description = "####{issue.id}"
+          issue2.save!
+          issue.description = "####{issue2.id}"
```
Comment (ulferts) [inline spec/helpers/application_helper_spec.rb:330]: isn't this one hash sign to much

**Comment by ulferts** [comment]
Cuke fails: cucumber features/work_packages/work_package_textile_link.feature:88

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (crijke) [inline app/helpers/application_helper.rb]: That's actually the thing that has been fixed: When replacing macros in the description of ticket A, '###B' will be replaced with '###A', but that will not be replaced, stopping the recursion.

File: spec/helpers/application_helper_spec.rb (hunk 2 above)
Comment (crijke) [inline spec/helpers/application_helper_spec.rb:330]: It's a bit confusing, but the last hash sign is for ruby code string enclosing  #{..}

**Comment by hschink** [comment]
Fortunately, your spec is failing: :smiley:

```
ApplicationHelper.textilizable Issue links Cyclic Description Links.ApplicationHelper.textilizable Issue links Cyclic Description Links doesn't replace description links with a cycle

stack level too deep (ActionView::Template::Error)
ActionView::Template::Error:
stack level too deep
     /home/jenkins/.rvm/gems/ruby-1.9.3-p327@c69fc841daee8954a65a8902c6292331/gems/activesupport-3.2.15/lib/active_support/notifications/instrumenter.rb:23
```

**Comment by hschink** [comment]
Internal CI is green :arrow_right_hook: ready to merge!

**Pull request merged**